Modules/_elementtree.c

```diff
@@ -2444,9 +2464,10 @@ _elementtree_TreeBuilder___init___impl(TreeBuilderObject *self,
 }
 
 static int
-treebuilder_gc_traverse(TreeBuilderObject *self, visitproc visit, void *arg)
+treebuilder_gc_traverse(PyObject *op, visitproc visit, void *arg)
 {
-    Py_VISIT(Py_TYPE(self));
+    Py_VISIT(Py_TYPE(op));
+    TreeBuilderObject *self = _TreeBuilder_CAST(op);
```

Comment:
I would prefer to keep self to reduce the number of changes:

```suggestion
    TreeBuilderObject *self = _TreeBuilder_CAST(op);
    Py_VISIT(Py_TYPE(self));
```